src/storage/datacore/restapi/custom/api.pm

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright 2024 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+package storage::datacore::restapi::custom::api;
+use strict;
+use warnings;
+use centreon::plugins::http;
+use centreon::plugins::statefile;
+use JSON::XS;
+use centreon::plugins::misc qw(empty);
+
+sub new {
+    my ($class, %options) = @_;
+    if (!defined($options{output})) {
+        print "Class Custom: Need to specify 'output' argument.\n";
+        exit 3;
+    }
+    if (!defined($options{options})) {
+        $options{output}->add_option_msg(short_msg => "Class Custom: Need to specify 'options' argument.");
+        $options{output}->option_exit();
+    }
+    my $self = {};
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'hostname:s' => { name => 'hostname' },
+        'port:s'     => { name => 'port', default => 443 },
+        'proto:s'    => { name => 'proto', default => 'https' },
+        'timeout:s'  => { name => 'timeout' },
+        'username:s' => { name => 'username' },
+        'password:s' => { name => 'password' }
+    });
+    $self->{output} = $options{output};
```

Comment:
I thibk the magical formula :
```$options{options}->add_help(package => __PACKAGE__, sections => 'API OPTIONS', once => 1);```
is also missing here

Reply:
what does it should do ?
I don't see any difference in the help generated with it.

Reply:
I don't see any difference in the help with it, what should change ?

Reply:
checked with Olivier : should have used "REST API OPTIONS", it should be the same as in the head1 of the pod document.